lib/repl.js

```diff
@@ -386,14 +386,15 @@ function REPLServer(prompt,
   self.bufferedCommand = '';
   self.lines.level = [];
 
-  function complete(text, callback) {
-    self.complete(text, callback);
-  }
+  // Figure out which "complete" function to use.
+  self.completer = (typeof options.completer === 'function')
+    ? options.completer
+    : complete.bind(this);
```

Comment:
@Fishrock123 It is because the completer function is passed immediately to the `Interface.completer` option, so if is removed, then the function will not have the correct binding when executed by the `readline` module.

The `apply` binding is for the case when the complete function is called directly through the `REPL` instance, such as the tests.

Is this still viable or maybe you can give me any idea on what change I need to make in order to be accepted?

 However, I will change this to `self` for consistency, it seems that I missed that.